src/main/scala/cromwell/engine/SymbolTable.scala

```diff
@@ -0,0 +1,34 @@
+package cromwell.engine
+
+import cromwell.binding.types.WdlType
+import cromwell.binding.{FullyQualifiedName, Call, WdlBinding, WdlValue}
+import scala.collection.mutable
+
+class SymbolTableEntry(val scope: String, val name: String, val iteration: Int, var wdlValue: Option[WdlValue], val wdlType: WdlType, input: Boolean) {
+  def isInput: Boolean = input
+
+  def isOutput: Boolean = !isInput
+}
+
+class SymbolTable(binding: WdlBinding, inputs: Map[FullyQualifiedName, WdlValue]) {
+  val table = mutable.Set[SymbolTableEntry]()
```

Comment:
Yeah... I guess I personally would have to be sold on it a bit more.  Some concerns I have which weren't immediately answered by a brief look at the Akka Persistence docs"

1.  It'd be nice to be able to query these data structures in a developer friendly way... If Akka persists things but in not a very query-able way or not meant to be queried that could be a draw back

2.  It seems to me like we'd want to tear down actors for completed workflows.  I know they're lightweight and all, but it doesn't seem scalable to me to leave them around for all of time because that's where our data is persisted.  If we ended up doing both Akka persistence AND database persistence separately I'd just say why even bother with Akka persistence?

3. User interfaces will have to get at this data somehow.  Maybe the user interface could send a message to the actor to get it's data but that runs into a problem with issue (2).  Also by using a database and designing a schema ourselves it allows for the possibility of third-party tools (like RedShirt is to Firehose) that provide UI and tooling.